### PR TITLE
Handle pktiohandler restarts and ports info to file

### PR DESCRIPTION
**Pull request by DanG100**

### CODE REVIEW

File: dataplane/saiserver/hostif.go [hunk 1]
```diff
@@ -57,7 +59,19 @@ type hostif struct {
...
+	remoteHostifs    map[uint64]*pktiopb.HostPortControlMessage
+	remoteClosers    []func()
+	remotePortReq    func(msg *pktiopb.HostPortControlMessage) error
+}
+
+func (hostif *hostif) Reset() {
+	hostif.trapIDToHostifID = map[uint64]uint64{}
+	hostif.groupIDToQueue = map[uint64]uint32{}
+	hostif.remoteHostifs = map[uint64]*pktiopb.HostPortControlMessage{}
+	for _, closeFn := range hostif.remoteClosers {
```
Comment (guoshiuan) [inline dataplane/saiserver/hostif.go:72]: Do we need to clear remoteClosers too, and then re-init?

File: dataplane/saiserver/hostif.go [hunk 2]
```diff
@@ -235,6 +249,8 @@ func (hostif *hostif) CreateHostif(ctx context.Context, req *saipb.CreateHostifR
 
 func (hostif *hostif) createRemoteHostif(ctx context.Context, req *saipb.CreateHostifRequest) (*saipb.CreateHostifResponse, error) {
 	id := hostif.mgr.NextID()
+	hostif.remoteMu.Lock()
```
Comment (guoshiuan) [inline dataplane/saiserver/hostif.go:252]: I think remoteMu is used to guard remoteHostifs[], should we move it closer to where remoteHostifs is acutally used?

**Review by guoshiuan — APPROVED** [review]

File: dataplane/saiserver/hostif.go (hunk 1 above)
Comment (DanG100) [reply dataplane/saiserver/hostif.go:72]: done

File: dataplane/saiserver/hostif.go (hunk 2 above)
Comment (DanG100) [reply dataplane/saiserver/hostif.go]: done

**Pull request merged**